src/ol/source/Vector.js

```diff
@@ -241,11 +241,11 @@ class VectorSource extends Source {
     this.idIndex_ = {};
 
     /**
-     * A lookup of features without id (keyed by getUid(feature)).
+     * A lookup of features by uid (using getUid(feature)).
      * @private
      * @type {!Object<string, import("../Feature.js").default<Geometry>>}
      */
-    this.undefIdIndex_ = {};
+    this.uidIndex_ = {};
```

Comment:
If you manage to change the color lookup for hit detection to array index, you should not need this change.

Although I like the idea of unconditionally indexing features by uid, I fear that we use too many uids if we do that. Before we decide to make this change, I'd like to see some tests with our vector source examples to see if we end up generating more uids after this change. It might well be that the features have a uid already, but I'm not sure about that.

Reply:
Tests done, see below